bin/common-modules.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/local/env bash
+
+#####
+# Use OPTIONS/LONGOPTS(LONGOPTIONS) to set additional parameters.
+# Resources:
+# - https://github.com/google/styleguide/blob/gh-pages/shellguide.md#s4.2-function-comments
+# - https://stackoverflow.com/a/29754866
+#####
+function parse_args() {
+  if [[ "$*" != "" ]]; then
+    ! getopt --test >/dev/null
+    if [[ ${PIPESTATUS[0]} -ne 4 ]]; then
+      err '`getopt --test` failed in this environment.'
+      exit 1
+    fi
+
+    OPTIONS=Al:c:
+    LONGOPTS=all,label:,cluster:
+
+    # - regarding ! and PIPESTATUS see above
+    # - temporarily store output to be able to check for errors
+    # - activate quoting/enhanced mode (e.g. by writing out “--options”)
+    # - pass arguments only via   -- "$@"   to separate them correctly
+    ! PARSED=$(getopt --options=$OPTIONS --longoptions=$LONGOPTS --name "$0" -- "$@")
+    if [[ ${PIPESTATUS[0]} -ne 0 ]]; then
+      exit 1
+    fi
+    eval set -- "$PARSED"
+    while true; do
+      case "$1" in
+        -A | --all)
+          ALL_OPT='true'
```

Comment:
I chose to make more explicit that it's an OPT(ION) and they are of course global.